core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -526,8 +534,19 @@ public static boolean useOpenSsl() {
         return useOpenSsl;
     }
 
+    /**
+     * Returns the {@link TlsEngineType} that will be used for processing TLS connections.
+     *
+     * <p>The default value of this flag is "openssl", which means the default {@link TlsEngineType} will
+     * be used. Specify the {@code -Dcom.linecorp.armeria.tlsEngineType=<jdk|openssl>} JVM option to override
+     * the default.</p>
+     */
+    public static TlsEngineType tlsEngineType() {
+        return TLS_ENGINE_TYPE;
```

Comment:
- We need to call `setUseOpenSslAndDumpOpenSslInfo()` when this method is called for the first time.
- We have to return `TlsEngineType.JDK` if `OpenSsl.isAvailable() == false`

Reply:
Thank you for the detailed comment. I've some following quetions.
- Because we are not able to do null checkings on `TLS_ENGINE_TYPE`(because it's always non-null), should we add a variable to check whether `setUseOpenSslAndDumpOpenSslInfo()` is previously called ?